test/functional/mining_basic.py

```diff
@@ -362,6 +363,12 @@ def test_block_max_weight(self):
             expected_msg=f"Error: Specified -blockmaxweight ({MAX_BLOCK_WEIGHT + 1}) exceeds consensus maximum block weight ({MAX_BLOCK_WEIGHT})",
         )
 
+    def test_height_in_locktime(self):
+        self.log.info("Sanity check generated blocks have their coinbase timelocked to their height.")
+        self.generate(self.nodes[0], 1, sync_fun=self.no_op)
+        block = self.nodes[0].getblock(self.nodes[0].getbestblockhash(), 2)
+        assert_equal(block["tx"][0]["locktime"], block["height"] - 1)
+        assert_equal(block["tx"][0]["vin"][0]["sequence"], MAX_SEQUENCE_NONFINAL)
```

Comment:
Maybe add something like

```python
        genblock = create_block(hashprev=int(block["hash"], 16), tmpl=dict(height=block["height"]+1), ntime=block["time"]+1)
        genblock.vtx[0].vin[0].nSequence = SEQUENCE_FINAL
        genblock.vtx[0].rehash()
        genblock.hashMerkleRoot = genblock.calc_merkle_root()
        genblock.solve()
        self.nodes[0].submitblock(genblock.serialize().hex())
        assert_equal(self.nodes[0].getbestblockhash(), genblock.hash)
```

to ensure there's explicit test coverage for blocks with a coinbase whose nSequence is final?

I believe with this PR there remains implicit test coverage via feature_taproot (`gen_test_vectors` uses `SEQUENCE_FINAL`) and via rpc_getblockstats (via the test data in data/rpc_getblockstats.json).

Reply:
It feels out of place in mining_basic, and the PR that introduces validation checks for nSequence/nLockTime in coinbase does check various combinations in feature_block, including this one. So i think it's better left there?